tests/core/test_input_validation.py

```diff
@@ -949,3 +950,27 @@ def test_validate_axes_orthogonal(bias_index):
     assert np.array_equal(axes, axes_left)
     with pytest.raises(ValueError, match=msg):
         validate_axes(axes_left, must_be_orthogonal=True)
+
+
+def test_check_padding():
+    valid_padding = [3, 0, 1, 2, 2, 1, 2, 4, 1, 2, 3, 8]
+    check_padding(valid_padding)
+
+    check_padding([])
+    check_padding([1, 0])
+
+    invalid_padding = [8, 0, 1, 2, 3, 4]
+    with pytest.raises(ValueError):
+        check_padding(invalid_padding)
+
+    invalid_padding = [3, 0, 1]
+    with pytest.raises(ValueError):
+        check_padding(invalid_padding)
+
+    invalid_padding = [3, 0]
+    with pytest.raises(ValueError):
+        check_padding(invalid_padding)
+
+    invalid_padding = [3]
+    with pytest.raises(ValueError):
+        check_padding(invalid_padding)
```

Comment:
`@pytest.mark.parametrize` makes the expression of the test code more concise.
```suggestion


@pytest.mark.parametrize("padding", [[3, 0, 1, 2, 2, 1, 2, 4, 1, 2, 3, 8], [], [1, 0]])
def test_check_valid_padding(padding):
    check_padding(padding)


@pytest.mark.parametrize("padding", [[8, 0, 1, 2, 3, 4], [3, 0, 1], [3, 0]])
def test_check_invalid_padding(padding):
    with pytest.raises(ValueError):
        check_padding(padding)
```